rust/arrow/src/array/array.rs

```diff
@@ -999,6 +1007,25 @@ impl StructArray {
     pub fn columns(&self) -> Vec<&ArrayRef> {
         self.boxed_fields.iter().collect()
     }
+
+    /// Return field names in this struct array
+    pub fn column_names(&self) -> Vec<&str> {
+        match self.data.data_type() {
+            Struct(fields) => fields
+                .iter()
+                .map(|f| f.name().as_str())
+                .collect::<Vec<&str>>(),
+            _ => panic!("Struct array's data type is not struct!"),
```

Comment:
if this is guaranteed to never happen, you could use `unreachable!()` instead?

Reply:
Thanks for this hint.